Classes/Database/Query/QueryGenerator.php

```diff
@@ -24,7 +24,7 @@ class QueryGenerator
     /**
      * @throws Exception
      */
-    public function getTreeList($id, $depth, $begin = 0, $permClause = ''): float|int|string
+    public function getTreeList($id, $depth, $begin = 0, $permClause = ''): int|string
```

Comment:
It returns INT, if there is just one numer available and it returns string, if there are multiple numbers available. Sure, this method is a copy of TYPO3 source and they have to be careful changing return types. But this one here is our own method, we can adopt it to our needs as we want. I prefer to set string as return type in general.

Another solution is:
return type array and you do a GeneralUtility::intExplode() here. In that case all floating values were migrated, too.